Library/Homebrew/cmd/update-report.rb

```diff
@@ -249,6 +249,20 @@ def updated?
   def migrate_tap_migration
     report[:D].each do |full_name|
       name = full_name.split("/").last
+      if tap == "caskroom/cask"
+        next unless (dir = HOMEBREW_REPOSITORY/"Caskroom"/name).exist?
+        next unless new_tap_name = tap.tap_migrations[name] # skip if formula is not in tap_migrations list.
+        new_tap = Tap.fetch(new_tap_name)
+        new_tap.install unless new_tap.installed?
+        ohai "#{name} has been moved to Homebrew Core. Installing #{name}..."
```

Comment:
You may need to run `brew link --force` to overwrite any symlinks that Homebrew Cask has created.